yt_dlp/networking/_helper.py

```diff
@@ -206,3 +207,54 @@ def wrapper(self, *args, **kwargs):
                 e.handler = self
             raise
     return wrapper
+
+
+def _socket_connect(ip_addr, timeout, source_address):
+    af, socktype, proto, canonname, sa = ip_addr
+    sock = socket.socket(af, socktype, proto)
+    try:
+        if timeout is not socket._GLOBAL_DEFAULT_TIMEOUT:
+            sock.settimeout(timeout)
+        if source_address:
+            sock.bind(source_address)
+        sock.connect(sa)
+        return sock
+    except socket.error:
+        sock.close()
+        raise
+
+
+def create_connection(address, timeout=socket._GLOBAL_DEFAULT_TIMEOUT, source_address=None, *, _create_socket_func=_socket_connect):
+    # This is to workaround _create_connection() from socket where it will try all
+    # address data from getaddrinfo() including IPv6. This filters the result from
+    # getaddrinfo() based on the source_address value.
+    # This is based on the cpython socket.create_connection() function.
+    # https://github.com/python/cpython/blob/main/Lib/socket.py#L810
```

Comment:
How about this?
```suggestion
    # Work around socket._create_connection() which tries all
    # addresses from getaddrinfo() including IPv6.
    # This filters the addresses based on the given source_address.
    # Based on: https://github.com/python/cpython/blob/main/Lib/socket.py#L810
```

Reply:
`socket.create_connection()`*, but yep that would make more sense in the new context